src/core/database/common.repository.ts

```diff
@@ -13,7 +14,26 @@ export class CommonRepository {
     return await this.db.checkDeletePermission(id, session);
   }
 
-  async deleteNode(objectOrId: { id: ID } | ID) {
-    await this.db.deleteNode(objectOrId);
+  async deleteNode(objectOrId: { id: ID } | ID, changeset?: ID) {
+    if (!changeset) {
+      await this.db.deleteNode(objectOrId);
+      return;
+    }
+    try {
+      const id = isIdLike(objectOrId) ? objectOrId : objectOrId.id;
+      await this.db
+        .query()
+        .match([node('node', 'BaseNode', { id })])
+        .match([node('changeset', 'Changeset', { id: changeset })])
+        .merge([
+          node('changeset'),
+          relation('out', 'rel', 'changeset'),
+          node('node'),
+        ])
+        .onCreate.setValues({ 'rel.active': true, 'rel.deleting': true })
```

Comment:
I don't think we need the `.onCreate`. If the relationship previously existed we still want to mark it as deleting.

Reply:
Then we need both `.onMatch` and `.onCreate`?

Reply:
I think they can just be two completely separate clauses